apps/web-giddh/src/app/actions/general/general.actions.ts

```diff
@@ -54,7 +56,68 @@ export class GeneralActions {
             switchMap((action: CustomActions) => this._companyService.getAllStates(action.payload)),
             map(resp => this.getAllStateResponse(resp)));
 
-    constructor(private action$: Actions, private _groupService: GroupService, private _accountService: AccountService, private _companyService: CompanyService) {
+    @Effect()
+    public updateIndexDb$: Observable<Action> = this.action$
+        .ofType(GENERAL_ACTIONS.UPDATE_INDEX_DB).pipe(
+            switchMap((action: CustomActions) => {
+                const payload: UpdateDbRequest = action.payload;
+                return this._dbService.getItemDetails(payload.uniqueName).pipe(map(itemData => ({itemData, payload})))
+            }),
+            switchMap(data => {
+                // debugger;
+                if (data.itemData && data.payload) {
+                    const payload = data.payload;
+                    const items = data.itemData;
+                    switch (payload.type) {
+                        case "accounts": {
+                            // debugger;
+                            const matchedIndex = items.aidata.accounts.findIndex(item => item && item.uniqueName && item.uniqueName === payload.oldUniqueName);
+                            if (matchedIndex > -1) {
+                                items.aidata.accounts = items.aidata.accounts.map(item => {
```

Comment:
Can this map be avoided as we are using `findIndex` to find the index of the element once the index is found we can directly update the `accounts` with the help of found index no need to iterate the list again